pkg/filter/stream/flowcontrol/callbacks.go

```diff
@@ -12,32 +12,52 @@ import (
 	"mosn.io/pkg/buffer"
 )
 
+var callbacksRegistry = make(map[string]Callbacks)
+
 // Callbacks defines the flow control callbacks
 type Callbacks interface {
 	Init()
-	ParseResource(ctx context.Context, headers types.HeaderMap, buf types.IoBuffer, trailers types.HeaderMap) *ParsedResource
+	ShouldIgnore(flowControlFilter *StreamFilter, ctx context.Context, headers types.HeaderMap, buf types.IoBuffer, trailers types.HeaderMap) bool
+	Prepare(ctx context.Context, headers types.HeaderMap, buf types.IoBuffer, trailers types.HeaderMap, trafficType base.TrafficType)
+	ParseResource(ctx context.Context, headers types.HeaderMap, buf types.IoBuffer, trailers types.HeaderMap, trafficType base.TrafficType) *ParsedResource
 	AfterBlock(flowControlFilter *StreamFilter, ctx context.Context, headers types.HeaderMap, buf types.IoBuffer, trailers types.HeaderMap)
 	AfterPass(flowControlFilter *StreamFilter, ctx context.Context, headers types.HeaderMap, buf types.IoBuffer, trailers types.HeaderMap)
+	Append(ctx context.Context, headers types.HeaderMap, buf types.IoBuffer, trailers types.HeaderMap)
 	Exit(filter *StreamFilter)
 	Enabled() bool
 }
 
-// ParsedResource contains the parsed resource wrapper and entry options.
+// ParsedResource contains the parsed Resource wrapper and entry options.
 type ParsedResource struct {
-	resource *base.ResourceWrapper
-	opts     []sentinel.EntryOption
+	Resource *base.ResourceWrapper
+	Opts     []sentinel.EntryOption
 }
 
 // DefaultCallbacks represents the default flow control filter implementation.
 type DefaultCallbacks struct {
 	config *Config
 }
 
+// RegisterCallbacks stores customized callbacks.
+func RegisterCallbacks(name string, cb Callbacks) {
+	callbacksRegistry[name] = cb
+}
+
+// GetCallbacksByName returns specified or default Callbacks.
+func GetCallbacksByConfig(conf *Config) Callbacks {
+	cb, ok := callbacksRegistry[conf.CallbackName]
```

Comment:
There may be concurrency conflict issues when dynamically updating the configuration. Could use `sync.map` to avoid it.